applications/Chat/coati/trainer/sft.py

```diff
@@ -38,94 +36,93 @@ def __init__(
         optim: Optimizer,
         lr_scheduler: _LRScheduler,
         train_dataloader: DataLoader,
-        eval_dataloader: DataLoader = None,
+        eval_dataloader: Optional[DataLoader] = None,
         max_epochs: int = 2,
         accumulation_steps: int = 8,
-        callbacks: List[Callback] = [],
     ) -> None:
         if accumulation_steps > 1 and isinstance(strategy, ColossalAIStrategy):
             from colossalai.booster.plugin import GeminiPlugin
             assert not isinstance(strategy.plugin, GeminiPlugin), \
                 "Accumulation steps are not supported in stage 3 of ColossalAI"
-        super().__init__(strategy, max_epochs, callbacks=callbacks)
-        self.train_dataloader = train_dataloader
+
+        super().__init__(
+            strategy, max_epochs,
+            model, optim, train_dataloader
+        )
+
         self.eval_dataloader = eval_dataloader
-        self.model = model
-        self.optimizer = optim
 
         self.accumulation_steps = accumulation_steps
-
         self.scheduler = lr_scheduler
 
-    def fit(self, logger, use_wandb: bool = False):
+    def _train(self, epoch: int):
+        self.model.train()
+        for batch_id, batch in enumerate(self.train_dataloader):
+
+            batch = to_device(batch, torch.cuda.current_device())
+            outputs = self.model(batch["input_ids"],
+                                 attention_mask=batch["attention_mask"],
+                                 labels=batch["labels"])
+
+            loss = outputs.loss
+
+            if loss >= 2.5 and is_rank_0():
```

Comment:
This is not reasonable. I think we can remove this.

Reply:
This is fixed in 9568185.